src/StoreApi/Schemas/V1/CartItemSchema.php

```diff
@@ -458,8 +458,19 @@ protected function get_item_data( $cart_item ) {
 		 * @param array $cart_item Cart item array.
 		 * @return array
 		 */
-		$item_data = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
-		return array_map( [ $this, 'format_item_data_element' ], $item_data );
+		$item_data           = apply_filters( 'woocommerce_get_item_data', array(), $cart_item );
+		$formatted_item_data = array_map( [ $this, 'format_item_data_element' ], $item_data );
+
+		// Remove empty arrays from the data.
+		$valid_item_data = array_filter(
+			$formatted_item_data,
+			function ( $item ) {
+				return ! empty( $item );
+			}
+		);
+
+		// Return only the values, so that this will be an object when it reaches the client.
+		return array_values( $valid_item_data );
```

Comment:
Did we always return values? this seems like a shift in the type of data returned?

Reply:
Previously the array started at index 0 and had sequential index numbers, now because we may filter some out we have to do this to reindex it to ensure the returned value appears as an array in JS